include/deal.II/base/grid_tools.h

```diff
@@ -0,0 +1,429 @@
+     *
+     * Here is an example that shows how this function is to be used
+     * in a concrete context. It is taken from the code that makes
+     * sure that the @p active_fe_index (a single unsigned integer) is
+     * transported from locally owned cells where one can set it in
+     * hp::DoFHandler objects, to the corresponding ghost cells on
+     * other processors to ensure that one can query the right value
+     * also on those processors:
+     * @code
+     *    auto pack
+     *    = [] (const typename dealii::hp::DoFHandler<dim,spacedim>::active_cell_iterator &cell) -> unsigned int
+     *    {
+     *      return cell->active_fe_index();
+     *    };
+     *
+     *    auto unpack
+     *      = [] (const typename dealii::hp::DoFHandler<dim,spacedim>::active_cell_iterator &cell,
+     *            const unsigned int                                                        &active_fe_index) -> void
+     *    {
+     *      cell->set_active_fe_index(active_fe_index);
+     *    };
+     *
+     *    parallel::GridTools::exchange_cell_data_to_ghosts<unsigned int, dealii::hp::DoFHandler<dim,spacedim>>
+     *        (dof_handler, pack, unpack);
+     * @endcode
+     *
+     * You will notice that the @p pack lambda function returns an `unsigned int`,
+     * not a `boost::optional<unsigned int>`. The former converts automatically
+     * to the latter, implying that data will always be transported to the
+     * other processor.
+     *
+     * (In reality, the @p unpack function needs to be a bit more
+     * complicated because it is not allowed to call
+     * DoFAccessor::set_active_fe_index() on ghost cells. Rather, the
+     * @p unpack function directly accesses internal data structures. But
+     * you get the idea -- the code could, just as well, have exchanged
+     * material ids, user indices, boundary indictors, or any kind of other
+     * data with similar calls as the ones above.)
+     */
+    template <typename DataType, typename MeshType>
+    void
+    exchange_cell_data_to_ghosts (const MeshType &mesh,
+                                  std::function<boost::optional<DataType> (const typename MeshType::active_cell_iterator &)> pack,
+                                  std::function<void (const typename MeshType::active_cell_iterator &, const DataType &)> unpack);
+
+    /**
+     * Compute a vector of bounding boxes which approximates the space occupied by
+     * the locally_owned cells. The number of bounding boxes (two or more) tries
+     * to catch the general shape of the space occupied by the locally_owned_cells,
+     * describing connected components and bigger shape variations.
+     *
+     * This function is meant to work only with a parallel::distributed::Triangulation
+     * which has been ordered following the Morton-Z curve order.
+     */
+
+    template < int dim, int spacedim>
+    std::vector<BoundingBox<spacedim>>
+                                    compute_locally_owned_bounding_box
```

Comment:
it looks like you need to run `astyle` indentation script

Reply:
Nah, this is just how astyle does it because it doesn't understand (yet) that `>>` is not an operator but two closing template arguments. You can get nicer indentation if you write it as `> >` (with a space).